### PR TITLE
Make `LR scheduler` stub complete, including `OneCycleLR` and class attributes.

### PR DESCRIPTION
**Pull request by sbrodehl**
This PR completes the stub file for lr scheduler and includes a previously missing scheduler, namely `OneCycleLR, and adds additional class attributes and methods for all lr scheduler.

### CODE REVIEW

**Comment by sbrodehl** [comment]
Stubs are still incomplete in the main branch, so this is not a stale PR.

File: torch/optim/lr_scheduler.pyi [hunk 1]
```diff
@@ -34,18 +51,61 @@ class SequentialLR(_LRScheduler):
...
+    total_size: Any = ...
+    step_ratio: Any = ...
+    mode: Any = ...
+    gamma: Any = ...
+    scale_fn: Any = ...
+    scale_mode: str = ...
+    cycle_momentum: Any = ...
+    base_momentums: Any = ...
+    max_momentums: Any = ...
+    base_lrs: Any = ...
```
Comment (jbschlosser) [inline torch/optim/lr_scheduler.pyi:92]: `base_lrs` is already defined in the base. Does this need to be here?

File: torch/optim/lr_scheduler.pyi [hunk 2]
```diff
@@ -34,18 +51,61 @@ class SequentialLR(_LRScheduler):
...
+    def __init__(self, optimizer: Optimizer, base_lr: Union[float, List[float]], max_lr: Union[float, List[float]], step_size_up: int = ..., step_size_down: Optional[int] = ..., mode: str = ..., gamma: float = ..., scale_fn: Optional[Callable[[float], float]] = ..., scale_mode: str = ..., cycle_momentum: bool = ..., base_momentum: float = ..., max_momentum: float = ..., last_epoch: int = ..., verbose: bool = ...) -> None: ...
 
 class CosineAnnealingWarmRestarts(_LRScheduler):
-    def __init__(self, optimizer: Optimizer, T_0: int=..., T_mult: int=..., eta_min: float=..., last_epoch: int=...) -> None: ...
+    T_0: int = ...
+    T_i: Any = ...
+    T_mult: Any = ...
+    eta_min: Any = ...
+    T_cur: Any = ...
+    last_epoch: Any = ...
```
Comment (jbschlosser) [inline torch/optim/lr_scheduler.pyi:101]: `last_epoch` is already defined in the base. Does this need to be here?

**Review by jbschlosser — COMMENTED** [review]
Thanks for the update! I've got a few questions below but otherwise looks fine

File: torch/optim/lr_scheduler.pyi [hunk 3]
```diff
@@ -2,21 +2,37 @@ from typing import Iterable, Any, Optional, Callable, Union, List
...
     def load_state_dict(self, state_dict: dict) -> None: ...
     def get_last_lr(self) -> List[float]: ...
     def get_lr(self) -> float: ...
-    def step(self, epoch: Optional[int]=...) -> None: ...
+    def step(self, epoch: Optional[int] = ...) -> None: ...
+    def print_lr(self, is_verbose: bool, group: dict, lr: float, epoch: Optional[int] = ...) -> None: ...
 
 class LambdaLR(_LRScheduler):
-    def __init__(self, optimizer: Optimizer, lr_lambda: Union[Callable[[int], float], List[Callable[[int], float]]], last_epoch: int=...) -> None: ...
+    optimizer: Optimizer = ...
```
[the remark this answers is not in the document]
Comment (sbrodehl) [reply torch/optim/lr_scheduler.pyi:18]: Right, not needed as you pointed out.

File: torch/optim/lr_scheduler.pyi [hunk 4]
```diff
@@ -2,21 +2,37 @@ from typing import Iterable, Any, Optional, Callable, Union, List
...
+    lr_lambdas: List[Callable[[int], float]] = ...
+    def __init__(self, optimizer: Optimizer, lr_lambda: Union[Callable[[int], float], List[Callable[[int], float]]], last_epoch: int = ..., verbose: bool = ...) -> None: ...
```
[the remark this answers is not in the document]
Comment (sbrodehl) [reply torch/optim/lr_scheduler.pyi:20]: `lr_lambda` is only a parameter of the `init` function, but not a member of the class.
Therefore, it shoult not be included.

File: torch/optim/lr_scheduler.pyi [hunk 5]
```diff
@@ -2,21 +2,37 @@ from typing import Iterable, Any, Optional, Callable, Union, List
...
+
+class MultiplicativeLR(_LRScheduler):
+    optimizer: Optimizer = ...
+    lr_lambdas: List[Callable[[int], float]] = ...
+    def __init__(self, optimizer: Optimizer, lr_lambda: Union[Callable[[int], float], List[Callable[[int], float]]], last_epoch: int = ..., verbose: bool = ...) -> None: ...
```
[the remark this answers is not in the document]
Comment (sbrodehl) [reply torch/optim/lr_scheduler.pyi:25]: It should not be included, same as `LambdaLR`.

File: torch/optim/lr_scheduler.pyi [hunk 6]
```diff
@@ -25,7 +41,8 @@ class LinearLR(_LRScheduler):
     def __init__(self, optimizer: Optimizer, start_factor: float=..., end_factor: float=..., total_iters: int=..., last_epoch: int=...) -> None: ...
```
[the remark this answers is not in the document]
Comment (sbrodehl) [reply torch/optim/lr_scheduler.pyi:41]: Yes!

File: torch/optim/lr_scheduler.pyi (hunk 1 above)
Comment (sbrodehl) [reply torch/optim/lr_scheduler.pyi:92]: No.

File: torch/optim/lr_scheduler.pyi (hunk 2 above)
Comment (sbrodehl) [reply torch/optim/lr_scheduler.pyi:101]: No.

**Comment by jbschlosser** [comment]
> Is this something we should be concerned in this PR?

I think it's outside of the scope of this PR, but thanks for bringing it up! The state of the LR schedulers is arguably poor, and unfortunately there is no owner currently dedicated to cleaning it up.

**Review by jbschlosser — APPROVED** [review]
